lib/exq/redis/job_queue.ex

```diff
@@ -75,7 +75,9 @@ defmodule Exq.Redis.JobQueue do
     score = Time.time_to_score(time)
 
     try do
-      case Connection.zadd(redis, scheduled_queue, score, job_serialized) do
```

Comment:
Possible to add duplicate message here?

Reply:
```
127.0.0.1:6379> ZADD scheduled 1 abc
(integer) 1
127.0.0.1:6379> ZRANGE scheduled 0 100
1) "abc"
127.0.0.1:6379> ZADD scheduled 1 abc
(integer) 0
127.0.0.1:6379> ZRANGE scheduled 0 100
1) "abc"
```
> The same element only exists a single time, no repeated elements are permitted

Calling ZADD multiple times itself is not a problem. Technically it's possible though to end up having multiple jobs

1. call ZADD (timeouts, but the operation succeeded)
2. another worker move the job from scheduled queue to work queue
3. retrying ZADD added the same job to scheduled queue again

The major risk is the time gap between 1 & 2. If the job is scheduled to run in 60 seconds in the future, less likely to happen. But if the job is scheduled to run in 1 second, this could happen.

Reply:
The ideal way to solve this problem is by not creating duplicate jobs under any conditions, this might require storing more information. My opinion is that the patch improves the current behavior at least. 

Currently, if there is a timeout
1) the job will get stuck in the in-process queue (on worker restart, it will get re-enqueued again)
2) if the operation succeeded on the Redis server, a duplicated job will get created when the worker gets restarted. One from scheduled-queue, one from the recovery of the in-process queue on worker restart